services/minespace-web/common/reducers/mineReducer.ts

```diff
@@ -1,13 +1,27 @@
+import { RootState } from "@/App";
 import * as actionTypes from "../constants/actionTypes";
 import { MINES } from "../constants/reducerTypes";
 import { createItemMap, createItemIdsArray } from "../utils/helpers";
-
+import { IMine, IMineComment, IMineDocument } from "@mds/common";
 /**
  * @file mineReducer.js
  * all data associated with new mine/existing mine records is handled witnin this reducer.
  */
+interface MineState {
+  mines: IMine;
+  mineIds: string[];
+  mineNameList: IMine[];
+  minesPageData: IMine;
+  mineGuid: any;
```

Comment:
Won't be able to avoid it entirely with these conversions, but it's probably best if we set as many `any` types to something more specific (ie. string)